integrationTests/relayers/ethToMultiversXWithChainSimulator_test.go

```diff
@@ -157,178 +162,467 @@ type keysHolder struct {
 	ethAddress common.Address
 }
 
-type testConfig struct {
-	isMintBurnOnMvX bool
-	isNativeOnMvx   bool
-	isMintBurnOnEth bool
-	isNativeOnEth   bool
+func TestRelayersShouldExecuteTransfers(t *testing.T) {
+	t.Run("ETH->MVX and back, ethNative = true, ethMintBurn = false, mvxNative = false, mvxMintBurn = true", testRelayersShouldExecuteTransfersEthToMVX(
+		argSimulatedSetup{
+			mvxHexIsMintBurn: hexTrue,
+			mvxHexIsNative:   hexFalse,
+			ethIsMintBurn:    false,
+			ethIsNative:      true,
+		},
+	))
+	t.Run("MVX->ETH, ethNative = false, ethMintBurn = true, mvxNative = true, mvxMintBurn = false", testRelayersShouldExecuteTransfersMVXToETH(
+		argSimulatedSetup{
+			mvxHexIsMintBurn: hexFalse,
+			mvxHexIsNative:   hexTrue,
+			ethIsMintBurn:    true,
+			ethIsNative:      false,
+		},
+	))
 }
 
-func TestTransfersBothWaysWithChainSimulator(t *testing.T) {
-	t.Run("Eth: native, MvX: mint & burn", func(t *testing.T) {
-		cfg := testConfig{
-			isMintBurnOnMvX: true,
-			isNativeOnMvx:   false,
-			isMintBurnOnEth: false,
-			isNativeOnEth:   true,
+func testRelayersShouldExecuteTransfersEthToMVX(argsSimulatedSetup argSimulatedSetup) func(t *testing.T) {
+	return func(t *testing.T) {
+		defer func() {
+			r := recover()
+			if r != nil {
+				require.Fail(t, "should have not panicked")
+			}
+		}()
+
+		argsSimulatedSetup.t = t
+		testSetup := prepareSimulatedSetup(argsSimulatedSetup)
+		defer testSetup.close(t)
+
+		checkESDTBalance(t, testSetup.testContext, testSetup.mvxChainSimulator, testSetup.mvxReceiverAddress, testSetup.mvxUniversalToken, "0", true)
+
+		createBatch(t, testSetup, batchProcessor.ToMultiversX)
+
+		interrupt := make(chan os.Signal, 1)
+		signal.Notify(interrupt, syscall.SIGINT, syscall.SIGTERM)
+		ethToMVXDone := false
+		mvxToETHDone := false
+
+		safeAddr, err := data.NewAddressFromBech32String(testSetup.mvxSafeAddress)
+		require.NoError(t, err)
+
+		// send half of the amount back to ETH
+		valueToSendFromMVX := big.NewInt(0).Div(mintAmount, big.NewInt(2))
+		initialSafeValue, err := testSetup.mvxChainSimulator.GetESDTBalance(testSetup.testContext, safeAddr, testSetup.mvxChainSpecificToken)
+		require.NoError(t, err)
+		initialSafeValueInt, _ := big.NewInt(0).SetString(initialSafeValue, 10)
+		expectedFinalValueOnMVXSafe := initialSafeValueInt.Add(initialSafeValueInt, feeInt)
+		expectedFinalValueOnETH := big.NewInt(0).Sub(valueToSendFromMVX, feeInt)
+		for {
+			select {
+			case <-interrupt:
+				require.Fail(t, "signal interrupted")
+				return
+			case <-time.After(time.Minute * 15):
+				require.Fail(t, "time out")
+				return
+			default:
+				isTransferDoneFromETH := checkESDTBalance(t, testSetup.testContext, testSetup.mvxChainSimulator, testSetup.mvxReceiverAddress, testSetup.mvxUniversalToken, mintAmount.String(), false)
+				if !ethToMVXDone && isTransferDoneFromETH {
+					ethToMVXDone = true
+					log.Info("ETH->MVX transfer finished, now sending back to ETH...")
+
+					sendMVXToEthTransaction(
```

Comment:
if you make the function `sendMVXToEthTransaction` part of the `testSetup` struct, 9 parameters from the function can be removed :) 

Valid for other helper functions that need at least one member from the testSetup struct

Reply:
updated all, good suggestion, cleaner code